### PR TITLE
US12-Task37 Setup a Login Page

### PR DESCRIPTION
**Pull request by AdvaitChirmule**
Screenshot:

image

### CODE REVIEW

File: react-ui/src/App.js [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+    <div className="mainbody">
+      <Form>
+        <Form.Group className="mb-3" controlId="formBasicEmail">
+          <Form.Label>Email Address: </Form.Label>
+          <Form.Control type="email" placeholder="Enter email" />
+        </Form.Group>
+
+        <Form.Group className="mb-3" controlId="formBasicPassword">
+          <Form.Label>Password: </Form.Label>
+          <Form.Control type="password" placeholder="Password" />
```
Comment (shahih25) [inline react-ui/src/App.js]: nit: Change placeholder to **Enter Password**

File: react-ui/src/index.css [hunk 2]
```diff
@@ -11,3 +11,13 @@ code {
   font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
     monospace;
 }
+<<<<<<< HEAD
```
Comment (shahih25) [inline react-ui/src/index.css]: This seems to be from merge conflict. Resolve it.

File: react-ui/src/App.js [hunk 3]
```diff
@@ -1,23 +1,31 @@
 import logo from './logo.svg';
```
Comment (shahih25) [inline react-ui/src/App.js]: nit: optimize imports

File: react-ui/public/index.html [hunk 4]
```diff
@@ -14,7 +14,9 @@
       manifest.json provides metadata used when your web app is installed on a
       user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN" crossorigin="anonymous">
     <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
+    <!doctype html>
```
Comment (shahih25) [inline react-ui/public/index.html]: I don't think we need this. Please try removing it and see it doesn't break anything.

**Comment by shahih25** [comment]
@AdvaitChirmule Is it possible to move the whole form in the middle of the screen?

File: react-ui/src/App.js [hunk 5]
```diff
@@ -1,23 +1,31 @@
...
+import Button from 'react-bootstrap/Button';
```
Comment (shahih25) [inline react-ui/src/App.js]: nit: _import { Button, Form, Row, Col } from "react-bootstrap";_

You can combine all the imports in one line from **react-bootstrap**.

File: react-ui/src/App.js [hunk 6]
```diff
@@ -1,23 +1,31 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="mainbody">
+      <Form>
+        <Form.Group className="mb-3" controlId="formBasicEmail">
+          <Form.Label>Email Address: </Form.Label>
```
Comment (shahih25) [inline react-ui/src/App.js]: It has to be **Username**. 

We can use both username and email to login afaik.

File: react-ui/src/App.js [hunk 7]
```diff
@@ -1,23 +1,31 @@
...
+          <Form.Control type="email" placeholder="Enter email" />
```
Comment (shahih25) [inline react-ui/src/App.js]: Change type to **text** to support both either username or email and change the placeholder to **Enter Username**.

File: react-ui/src/App.js [hunk 8]
```diff
@@ -1,23 +1,31 @@
...
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div className="mainbody">
+      <Form>
+        <Form.Group className="mb-3" controlId="formBasicEmail">
```
Comment (shahih25) [inline react-ui/src/App.js]: nit: _controlId="formBasicUsername"_ or _controlId="formUsername"_

**Comment by AdvaitChirmule** [comment]
> @AdvaitChirmule Is it possible to move the whole form in the middle of the screen?

Yes, it is done

**Review by shahih25 — APPROVED** [review]
lgtm

File: react-ui/src/App.js (hunk 1 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

File: react-ui/src/index.css (hunk 2 above)
Comment (AdvaitChirmule) [reply react-ui/src/index.css]: Done

File: react-ui/src/App.js (hunk 3 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

File: react-ui/public/index.html (hunk 4 above)
Comment (AdvaitChirmule) [reply react-ui/public/index.html]: Done

File: react-ui/src/App.js (hunk 5 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

File: react-ui/src/App.js (hunk 6 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

File: react-ui/src/App.js (hunk 7 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

File: react-ui/src/App.js (hunk 8 above)
Comment (AdvaitChirmule) [reply react-ui/src/App.js]: Done

**Pull request merged**